node/src/components/storage.rs

```diff
@@ -1718,7 +1706,19 @@ impl StorageInner {
             None => return Ok(None),
             Some(block_header_with_metadata) => block_header_with_metadata,
         };
-        if let Some(block_body) = self.get_body_for_block_header(tx, &block_header)? {
+        let (maybe_block_body, _) = get_body_for_block_header(
+            tx,
+            &block_header,
+            &Databases {
+                block_body_v1_db: self.block_body_v1_db,
+                block_body_v2_db: self.block_body_v2_db,
+                deploy_hashes_db: self.deploy_hashes_db,
+                transfer_hashes_db: self.transfer_hashes_db,
+                proposer_db: self.proposer_db,
+            },
+            self.verifiable_chunked_hash_activation,
+        );
```

Comment:
Since it's used in multiple places, that could still be extracted into a method (even with the same name).

Reply:
Sorry, that's not what I meant! there's no reason to use a private getter for that field!
I meant keeping the _method_ `self.get_body_for_block_header(tx, &block_header)` (which you removed), that would then call the _function_ `get_body_for_block_header`. Sorry for the confusion!

Reply:
I would not be able to use `self.get_body_for_block_header(tx, &block_header)` in the `StorageInner::new`, that's why I changed it to free function. The previous implementation was retrieving block in the constructor "manually" (probably because of lacking the `self`) and I wanted to keep the body retrieval code in one place.

I'm open to discussion, though.

Reply:
Sure thing, the revert is here: cd9aee38360eef70ac22c6f787c02a1f1630300f